isis/src/qisis/apps/ipce/IpceMainWindow.cpp

```diff
@@ -571,12 +572,47 @@ namespace Isis {
       QString projName = project->name();
       QString t0String=QString::number(t0);
 
-      if (!project->projectRoot().contains("tmpProject") &&
+      if (!project->isTemporaryProject() &&
           !projectPaths.contains( project->projectRoot())) {
         globalSettings.setValue(t0String+"%%%%%"+projName,project->projectRoot());
       }
     }
     globalSettings.endGroup();
+    globalSettings.sync();
+  }
+
+
+  /**
+   * Write the window positioning and state information out to a
+   * config file. This allows us to restore the settings when we
+   * create another main window (the next time this program is run).
+   *
+   * The state will be saved in the currently loaded project's root.
+   *
+   * @param[in] project Pointer to the project that is currently loaded (default is "Project")
+   *
+   * @internal
+   *   @history 2016-11-09 Ian Humphrey - Settings are now written according to the loaded project.
+   *                           References #4358.
+   *   @history 2017-10-17 Tyler Wilson Added a [recent projects] group for the saving and
+   *                           restoring of recently opened projects.  References #4492.
+   */
+  void IpceMainWindow::writeSettings(Project *project) {
+
+    // Ensure that we are not using a NULL pointer
+    if (!project) {
+      QString msg = "Cannot write settings with a NULL Project pointer.";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
+    QSettings projectSettings(FileName(project->newProjectRoot() + "/ipce.config").expanded(),
+        QSettings::NativeFormat);
+
+    projectSettings.setValue("geometry", QVariant(geometry()));
+    projectSettings.setValue("windowState", saveState());
+    projectSettings.sync();
+
+    //TODO Do we really need this? Isn't this a global setting? Can the user even change this?
```

Comment:
I don't know the answer to this "todo" but is there any chance it's been answered since the time this PR was put in?

Reply:
Yes, the user can change the maxThreadCount, so I think this should remain as a global setting.